assets/src/edit-story/components/richText/styleManipulation.js

```diff
@@ -0,0 +1,319 @@
+      ? shouldSetStyle([foundMatch])
+      : foundMatch === NONE;
+
+    // If so, add to list
+    if (willAddStyle) {
+      const styleToAdd = getStyleToSet ? getStyleToSet([foundMatch]) : prefix;
+      inlineStyles = inlineStyles.add(styleToAdd);
+    }
+
+    // Finally apply to style override
+    const newState = EditorState.setInlineStyleOverride(
+      editorState,
+      inlineStyles
+    );
+    return newState;
+  }
+
+  const matchingStyles = getPrefixStylesInSelection(editorState, prefix);
+
+  // First remove all old styles natching prefix
+  // (except NONE, it's not actually a style)
+  const stylesToRemove = matchingStyles.filter((s) => s !== NONE);
+  const strippedContentState = stylesToRemove.reduce(
+    (contentState, styleToRemove) =>
+      Modifier.removeInlineStyle(
+        contentState,
+        editorState.getSelection(),
+        styleToRemove
+      ),
+    editorState.getCurrentContent()
+  );
+
+  // Should we add a style to everything now?
+  // If no function is given, we simply add the style if any
+  // character did not have a match (NONE is in the list)
+  const willSetStyle = shouldSetStyle
+    ? shouldSetStyle(matchingStyles)
+    : matchingStyles.includes(NONE);
+
+  if (!willSetStyle) {
+    // we're done!
+    return applyContent(editorState, strippedContentState);
+  }
+
+  // Add style to entire selection
+  // If no function is given, we simple add the prefix as a style
+  const styleToSet = getStyleToSet ? getStyleToSet(matchingStyles) : prefix;
+  const newContentState = Modifier.applyInlineStyle(
+    strippedContentState,
+    editorState.getSelection(),
+    styleToSet
+  );
+
+  return applyContent(editorState, newContentState);
+}
+
+// convert a set of weight styles to a set of weights
+function getWeights(styles) {
+  return styles.map((style) =>
+    style === NONE ? NORMAL_WEIGHT : styleToWeight(style)
+  );
+}
+
+export function isBold(editorState) {
+  const styles = getPrefixStylesInSelection(editorState, WEIGHT);
+  const weights = getWeights(styles);
+  const allIsBold = weights.every((w) => w >= SMALLEST_BOLD);
+  return allIsBold;
+}
+
+export function toggleBold(editorState, flag) {
+  // if flag set, use flag
+  // otherwise if any character has weight less than SMALLEST_BOLD,
+  // everything should be bolded
+  const shouldSetBold = (styles) =>
+    typeof flag === 'boolean'
+      ? flag
+      : getWeights(styles).some((w) => w < SMALLEST_BOLD);
+
+  // if flag set, toggle to either 400 or 700,
+  // otherwise if setting a bold, it should be the boldest current weight,
+  // though at least DEFAULT_BOLD
+  const getBoldToSet = (styles) =>
+    typeof flag === 'boolean'
+      ? weightToStyle(flag ? DEFAULT_BOLD : NORMAL_WEIGHT)
+      : weightToStyle(
+          Math.max.apply(null, [DEFAULT_BOLD].concat(getWeights(styles)))
+        );
+
+  return togglePrefixStyle(editorState, WEIGHT, shouldSetBold, getBoldToSet);
+}
+
+export function getFontWeight(editorState) {
```

Comment:
Yes, everything in this file works on the current selection as represented by the editor state selection object. But this will get reworked too - to be split in verticals rather than horizontals (so there'e be one file for all the logic for `italic`, another for all `fontWeight`/`bold`, etc).

Reply:
Ok. And currently the full-text updates are `editorState.forceSelection(selectAll)`, right?

Reply:
Yes, actually the full-text updates are roughly `content = export(toggleBold(selectAll(import(content))));` as there is no actual editor to invoke these changes on - [this happens in `htmlManipulation`](https://github.com/google/web-stories-wp/pull/1323/files/47df628514fefc5c55a10424e0c0e1585edd6520#diff-4bb929db50ae52c99fcd00b92fa60ec7R38-R49).